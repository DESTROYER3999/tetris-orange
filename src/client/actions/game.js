import socket from "../services/socket-api";
import {
  KEY_PRESSED,
  UPDATE_SPECTRES,
  UPDATE_GAME_MODE
} from "../../constants/actionTypes";

export const handleKeyPress = e => {
  socket.emit(KEY_PRESSED, { code: e.code });
};

export const updateSpectres = spectres => ({
  type: UPDATE_SPECTRES,
  spectres
});

export const updateGameMode = mode => {
  return () => {
    socket.emit(UPDATE_GAME_MODE, mode);
  };
};
