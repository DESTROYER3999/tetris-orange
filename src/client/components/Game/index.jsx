import { connect } from "react-redux";
import GameSub from "./subcomponent";

const mapStateToProps = state => {
  return {
    mode: state.room.mode
  };
};

const Game = connect(
  mapStateToProps,
  null
)(GameSub);

export default Game;
