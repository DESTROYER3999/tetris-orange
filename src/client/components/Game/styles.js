import styled from "styled-components";

export const StyledContainer = styled.div`
  padding: 1vh;
  flex: 1;
  display: flex;
`;

export const StyledColumnContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex-basis: 20%;
  height: 86vh;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 0px 3px 3px 0px;
  border: 3px solid white;
  align-items: center;
  .value {
    font-size: 2vh;
  }
`;

export const StyledLeftItem = styled.div`
  flex: 2;
  padding: 1vh;
  margin: 5px;
  background-color: rgba(0, 0, 0, 0.3);
  border-radius: 2px 2px 2px 60px;
  display: flex;
  justify-content: space-around;
  align-items: center;
`;

export const StyledRightItem = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
`;

export const StyledLabel = styled.div`
  font-size: 3vh;
  padding: 1vh;
`;
