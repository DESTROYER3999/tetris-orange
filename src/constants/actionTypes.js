export const JOIN_ROOM = "JOIN_ROOM";
export const UPDATE_ROOM = "UPDATE_ROOM";
export const LEAVE_ROOM = "LEAVE_ROOM";
export const LOBBY_ROOM = "LOBBY_ROOM";

export const NEW_PLAYER = "NEW_PLAYER";
export const DISCONNECT_PLAYER = "DISCONNECT_PLAYER";
export const UPDATE_PLAYER = "UPDATE_PLAYER";
export const UPDATE_PLAYERS_LIST = "UPDATE_PLAYERS_LIST";
export const DELETE_PLAYER = "DELETE_PLAYER";

export const HANDLE_HASH = "HANDLE_HASH";

export const UPDATE_BOARD = "UPDATE_BOARD";
export const UPDATE_GAME = "UPDATE_GAME";
export const UPDATE_GAME_MODE = "UPDATE_GAME_MODE";
export const UPDATE_SPECTRES = "UPDATE_SPECTRES";

export const DISPLAY_TOAST = "DISPLAY_TOAST";

export const CONNECTION = "connection";
export const DISCONNECT = "disconnect";

export const ADD_CHAT_MESSAGE = "ADD_CHAT_MESSAGE";
export const NEW_CHAT_MESSAGE = "NEW_CHAT_MESSAGE";

export const KEY_PRESSED = "KEY_PRESSED";
