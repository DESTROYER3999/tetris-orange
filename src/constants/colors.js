export const MALUS = "#3d3d3d";

export const VIVID_BLUE = "#143cdc";

export const CYAN_BLUE = "#14d4dc";

export const CRIMSON = "#dc143c";

export const PURE_ORANGE = "#ff8b00";

export const LIME_GREEN = "#3cdc14";

export const PURE_VIOLET = "#b600ff";

export const PURE_YELLOW = "#fff400";

export const DARK_CYAN = "#375f64";

export const GREEN_CHAT = "#73F054";

export const RED_CHAT = "#ff3366";
