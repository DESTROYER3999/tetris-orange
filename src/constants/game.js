export const SOLO = "Solo";
export const BATTLEROYAL = "Battleroyal";
export const MAX_PLAYER_SOLO = 1;
export const MAX_PLAYER_BATTLEROYAL = 5;
export const BOARD_HEIGHT = 20;
export const BOARD_WIDTH = 10;
