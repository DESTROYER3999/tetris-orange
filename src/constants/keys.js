export const enumKeys = {
  ARROW_LEFT: "ArrowLeft",
  ARROW_RIGHT: "ArrowRight",
  ARROW_DOWN: "ArrowDown",
  ARROW_UP: "ArrowUp",
  SPACE: "Space",
  ENTER: "Enter"
};
