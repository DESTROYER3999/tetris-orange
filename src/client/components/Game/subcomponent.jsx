import React from "react";
import Board from "../Board";
import GameInfo from "../GameInfo";
import {
  StyledContainer,
  StyledLeftItem,
  StyledRightItem,
  StyledColumnContainer
} from "./styles";
import Spectres from "../Spectres";
import Chat from "../Chat";
import NextPieces from "../NextPieces";
import Progression from "../Progression";
import { SOLO } from "../../../constants/game";
import PropTypes from "prop-types";

const GameSub = ({ mode }) => {
  const displayProgression = () => {
    if (mode === SOLO) return <Progression />;
  };
  return (
    <StyledContainer id="game">
      <StyledLeftItem id="left-item">
        <Spectres />
        <Board />
        <StyledColumnContainer>
          <NextPieces />
          {displayProgression()}
        </StyledColumnContainer>
      </StyledLeftItem>
      <StyledRightItem id="right-item">
        <GameInfo />
        <Chat />
      </StyledRightItem>
    </StyledContainer>
  );
};

GameSub.propTypes = {
  mode: PropTypes.string.isRequired
};

export default GameSub;
