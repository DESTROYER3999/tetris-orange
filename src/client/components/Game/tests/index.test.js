import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import configureStore from "redux-mock-store";
import Game from "../index";
import socket from "../../../services/socket-api";
import { SOLO } from "../../../../constants/game";

const mockStore = configureStore();
const initialState = {
  room: { mode: SOLO }
};

const store = mockStore(initialState);

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

describe("<Game />", () => {
  test("Game rendering with store without crashing", () => {
    const wrapper = shallow(<Game store={store} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
