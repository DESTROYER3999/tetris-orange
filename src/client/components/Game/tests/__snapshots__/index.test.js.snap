// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> Game rendering with store without crashing 1`] = `
<ContextProvider
  value={null}
>
  <GameSub
    dispatch={[Function]}
    mode="Solo"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
